pytorch_blade/torch_blade/quantization/prepare_data.py

```diff
@@ -195,28 +196,28 @@ def get_calib_data_for_each_group(self):
             all_data.append(data[1:])
         return all_data
 
-    def save_data(self, data):
-        tmp_dir = tempfile.mkdtemp(prefix="torch_blade")
-        file_path = os.path.join(tmp_dir, "calibration_data.pt")
-        torch.save(data, file_path)
-        return file_path
-
     def prepare(self):
         self.make_fusion_group_executable()
         self.insert_observer()
         all_data = self.get_calib_data_for_each_group()
-        file_path = self.save_data(all_data)
-        return file_path
+        return all_data
 
 
+@contextmanager
 def get_calib_file_for_each_group(c_module):
+    calib_data_for_all_fusion_group = None
     trt_calibration_data = Config.get_current_context_or_new().quantization_calibration_data
-    if not trt_calibration_data:
-        return None
-    try:
-        data_preparer = DataPreparer(c_module, trt_calibration_data)
-        file_path = data_preparer.prepare()
-        return file_path
-    except Exception as e:
-        logger.warning(f"Unable to get calib file for each graph group due to {e}.")
-        return None
+    if trt_calibration_data is not None:
+        try:
+            data_preparer = DataPreparer(c_module, trt_calibration_data)
+            calib_data_for_all_fusion_group = data_preparer.prepare()
+        except Exception as e:
+            logger.warning(f"Unable to get calib file for each graph group due to {e}.")
+
+    with tempfile.TemporaryDirectory(prefix="torch_blade") as tmp_dirname:
```

Comment:
Nit: How about making the tmp_dir implicit?
```
with tempfile.TemporaryDirectory(prefix=".torch_blade") as tmp_dirname:
```

Reply:
done